### ISSUE TITLE
Connection failed/ Service not recognized/ clearing fetch info/ slow-fast

### ISSUE THREAD

**Issue by trippinshrumes** [d8ahazard/Phlex#235]
I'm running this on a raspberry Pi B+ device. 

I'm using the apache method

for informational purposes;
The fastest the cpu has been throttled is about 66pct out of 400 
The ram has only been at around 80MB out of 927
Reads at 300MB MAX --- utilizing nomore than 50-100K during use
writes at 90MB MAX -- utilizing no more than 20-80k during use
ALL dependencies are installed and there are no errors in the error.log for apache
Bandwidth to outside world is 100/5
Bandwith internally is gigabit, although I think the pi only supports 100Mb
I've reinstalled with the newest version from the git today 12-2-17 approx 7pm central standard time

With these tests I've performed, I believe the PI is certainly strong enough to run this. So lets get down to the issues at hand. 

This may be extra details, although you never know what could help.

Installed it on the pi a few weeks ago before my google home mini even arrived. When I went to connect it back up today I noticed it was 30 commits behind. hit the auto update, then it said it was up to date. I immediately got to setting it up with couchpotato and sonarr. I was able to navigate the menu internally at a fast speed, although it didn't appear that any information was refreshing. It saw my Main server as well as the plex cloud one minute, and didn't the next. I attempted to use the flex tv and told it to play s01 e01 weeds (nothing happened), it responded that it was going to try to play that for me, everything looked fine on the gui main screen for phlex, it showed all info of the show etc., the logs showed that it had an issue connecting to the server I did not document it at that time. I also attempted to use the no plex direct option, the servers dissapeard briefly and came back, flex tv did the same thing as earlier. Clicking test on my plex servers as well as sonarr/couchpotato would give me service unrecognized error. I decided to completely reinstall everything.

Purged and reinstalled top to bottom

This time working much better, with default settings seeing my servers immediately and not delayed, was able to put in couchpotato info and tested and set profile with success,then I went and set up sonarr, and it says can't connect to server, I verified the information was correct 3 times and even copy pasted from phlex to make 100 pct I wasn't missing a syntax error. Tried use flextv play s01e01 of weeds, it said it would try to play it for me (didn't work). Looking at phlex gui it said the episode wasn't available would I like it to download. I went in to plex servers and with my server and plex cloud the test fails to connect. If I change the option to no plex direct, then the servers line stays blank, although testing will repeatedly tell me whichever server I was clicked on last failed to connect. 

I restarted the Pi, went into fetchers to set up sonarr, and couch potato was disabled, enabling it showed all the settings were cleared. I put the settings back in but test said service unavailable. Went to test plex server connection and it said service unavailable, although when I told phlextv to play s01e01 weeds, it said "i'll try to play s01e01 weeds for you" (didn't work). The gui showed that it was going to play it and had pulled the info of the episode etc. There was some errors in the log again, these are as follows:

```
Sat, 02 Dec 2017 21:10:28 -0600api.php::setSessionVariables
No DVR found, checking for available devices.
Sat, 02 Dec 2017 21:10:28 -0600api.php
INCOMING REQUEST!!
Sat, 02 Dec 2017 21:10:26 -0600api.php::testConnection
Function fired, testing connection for Couch
Sat, 02 Dec 2017 21:10:25 -0600api.php::setSessionVariables
No DVR found, checking for available devices.
Sat, 02 Dec 2017 21:10:25 -0600api.php::require_once
INCOMING REQUEST!!
Sat, 02 Dec 2017 21:03:53 -0600api.php::playMedia
Playback Result: 
{
  "url": "http://IP:32400/player/playback/playMedia?key=%2Flibrary%2Fmetadata%2F39242&offset=0&machineIdentifier=10a3949c30bfe5e3d8105af468956e4699f0b473&protocol=http&address=IP&port=32400&containerKey=%2FplayQueues%2F1165%3Fown%3D1%26window%3D200&token=transient-49a5bf65-5d44-430c-92f4-57870e98e170&commandID=3",
  "status": "error"
}
Sat, 02 Dec 2017 21:03:53 -0600api.php::playMediaRelayed
Playback URL is http://176.X.XX.XXX:32400/player/playback/playMedia?key=%2Flibrary%2Fmetadata%2F39242&offset=0&machineIdentifier=10a3949c30bfe5e3d8105af468956e4699f0b473&protocol=http&address=176.X.XX.XXX&port=32400&containerKey=%2FplayQueues%2F1165%3Fown%3D1%26window%3D200&token=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&commandID=3
Sat, 02 Dec 2017 21:03:53 -0600util.php::protectURL
Parts: ["IP"]
Sat, 02 Dec 2017 21:03:53 -0600util.php::curlGet
Unexpected HTTP code: 500, URL: http://IP:32400/player/playback/playMedia?key=%2Flibrary%2Fmetadata%2F39242&offset=0&machineIdentifier=10a3949c30bfe5e3d8105af468956e4699f0b473&protocol=http&address=IP&port=32400&containerKey=%2FplayQueues%2F1165%3Fown%3D1%26window%3D200&token=transient-49a5bf65-5d44-430c-92f4-57870e98e170&commandID=3
Sat, 02 Dec 2017 21:03:52 -0600api.php
Invalid API Token, forcing logout.
Sat, 02 Dec 2017 21:03:52 -0600api.php
ERROR: Unauthenticated access detected. Originating IP
Sat, 02 Dec 2017 21:03:52 -0600api.php
Sorry, couldn't validate user.
Sat, 02 Dec 2017 21:03:52 -0600api.php
INCOMING REQUEST!!
Sat, 02 Dec 2017 21:03:49 -0600api.php::queueMedia
URL is http://176.X.XX.XXX:32400/playQueues?type=video&uri=library://0ab5c237-5ea5-425b-abbb-7a6785c47932/item/%2Flibrary%2Fmetadata%2F39242&shuffle=0&repeat=0&includeChapters=1&own=1&X-Plex-Client-Identifier=2e8489919f22b7956f402e64&X-Plex-Product=Phlex&X-Plex-Version=1.0.0&X-Plex-Platform=Web&X-Plex-Platform-Version=1.0.0&X-Plex-Device=PhlexWeb&X-Plex-Device-Name=Phlex&X-Plex-Device-Screen-Resolution=1520x707,1680x1050,1920x1080&X-Plex-Token=XXXXXXXXXXXXXXXXXXXX
Sat, 02 Dec 2017 21:03:49 -0600api.php::queueMedia
URL is http://176.X.XX.XXX:32400/library/metadata/39242?X-Plex-Token=XXXXXXXXXXXXXXXXXXXX
Sat, 02 Dec 2017 21:03:49 -0600api.php::playMedia
Function fired!
Sat, 02 Dec 2017 21:03:49 -0600api.php::returnAssistantSpeech
JSON out is 
{
  "speech": "As you command, Playing  You Can't Miss the Bear.",
  "contextOut": [
    {
      "name": "yes",
      "lifespan": 2,
      "parameters": []
    }
  ],
  "data": {
    "google": {
      "expectUserResponse": false,
      "isSsml": false,
      "noInputPrompts": [],
      "richResponse": {
        "items": [
          {
            "simpleResponse": {
              "textToSpeech": "As you command, Playing  You Can't Miss the Bear.",
              "displayText": "As you command, Playing  You Can't Miss the Bear."
            }
          }
        ],
        "suggestions": []
      }
    }
  }
}
Sat, 02 Dec 2017 21:03:49 -0600api.php::returnAssistantSpeech
Not displaying card for Weeds
Sat, 02 Dec 2017 21:03:49 -0600api.php::parseApiCommand
Final Media Result: 
{
  "ratingKey": "39242",
  "key": "/library/metadata/39242",
  "parentRatingKey": "39241",
  "studio": "Showtime",
  "type": "episode",
  "title": "You Can't Miss the Bear",
  "parentKey": "/library/metadata/39241",
  "grandparentTitle": "Weeds",
  "parentTitle": "Season 1",
  "contentRating": "TV-MA",
  "summary": "In the series debut, housewife Nancy Botwin struggles to deal with the sudden death of her husband, while her best friend and neighbour Celia Hodes learns a devastating secret when she tries to catch her daughter in the act with Nancy's son.",
  "index": "1",
  "parentIndex": "1",
  "rating": "7.3",
  "year": "2005",
  "thumb": "http://IP:32400/photo/:/transcode?width=1920&height=1920&minSize=1&url=%2Flibrary%2Fmetadata%2F39242%2Fthumb%2F1499256826%3FX-Plex-Token%3DRzzoocB1JbsvaQx1dCVR&X-Plex-Token=RzzoocB1JbsvaQx1dCVR",
  "art": "http://IP:32400/photo/:/transcode?width=1920&height=1920&minSize=1&url=%2Flibrary%2Fmetadata%2F39142%2Fart%2F1499256812%3FX-Plex-Token%3DRzzoocB1JbsvaQx1dCVR&X-Plex-Token=RzzoocB1JbsvaQx1dCVR",
  "parentThumb": "/library/metadata/39241/thumb/1499256811",
  "grandparentThumb": "/library/metadata/-1/thumb/1499256086",
  "grandparentArt": "/library/metadata/-1/art/1499256086",
  "duration": "1901942",
  "originallyAvailableAt": "2005-08-07",
  "addedAt": "1499256086",
  "updatedAt": "1499256826"
}
Sat, 02 Dec 2017 21:03:49 -0600api.php::fetchInfo
Winner_art: /library/metadata/39142/art/1499256812
Sat, 02 Dec 2017 21:03:49 -0600api.php::fetchInfo
Winner_thumb: /library/metadata/39242/thumb/1499256826
Sat, 02 Dec 2017 21:03:49 -0600api.php::fetchInfo
We have a winner: 
{
  "ratingKey": "39242",
  "key": "/library/metadata/39242",
  "parentRatingKey": "39241",
  "studio": "Showtime",
  "type": "episode",
  "title": "You Can't Miss the Bear",
  "parentKey": "/library/metadata/39241",
  "grandparentTitle": "Weeds",
  "parentTitle": "Season 1",
  "contentRating": "TV-MA",
  "summary": "In the series debut, housewife Nancy Botwin struggles to deal with the sudden death of her husband, while her best friend and neighbour Celia Hodes learns a devastating secret when she tries to catch her daughter in the act with Nancy's son.",
  "index": "1",
  "parentIndex": "1",
  "rating": "7.3",
  "year": "2005",
  "thumb": "/library/metadata/39242/thumb/1499256826",
  "art": {
    "0": "/library/metadata/39142/art/1499256812"
  },
  "parentThumb": "/library/metadata/39241/thumb/1499256811",
  "grandparentThumb": "/library/metadata/-1/thumb/1499256086",
  "grandparentArt": "/library/metadata/-1/art/1499256086",
  "duration": "1901942",
  "originallyAvailableAt": "2005-08-07",
  "addedAt": "1499256086",
  "updatedAt": "1499256826"
}
Sat, 02 Dec 2017 21:03:49 -0600api.php::fetchNumberedTVItem
Matching episode: 
{
  "ratingKey": "39242",
  "key": "/library/metadata/39242",
  "parentRatingKey": "39241",
  "studio": "Showtime",
  "type": "episode",
  "title": "You Can't Miss the Bear",
  "parentKey": "/library/metadata/39241",
  "grandparentTitle": "Weeds",
  "parentTitle": "Season 1",
  "contentRating": "TV-MA",
  "summary": "In the series debut, housewife Nancy Botwin struggles to deal with the sudden death of her husband, while her best friend and neighbour Celia Hodes learns a devastating secret when she tries to catch her daughter in the act with Nancy's son.",
  "index": "1",
  "parentIndex": "1",
  "rating": "7.3",
  "year": "2005",
  "thumb": "/library/metadata/39242/thumb/1499256826",
  "art": {
    "0": "/library/metadata/39142/art/1499256812"
  },
  "parentThumb": "/library/metadata/39241/thumb/1499256811",
  "grandparentThumb": "/library/metadata/-1/thumb/1499256086",
  "grandparentArt": "/library/metadata/-1/art/1499256086",
  "duration": "1901942",
  "originallyAvailableAt": "2005-08-07",
  "addedAt": "1499256086",
  "updatedAt": "1499256826"
}
Sat, 02 Dec 2017 21:03:48 -0600api.php::fetchNumberedTVItem
URL is http://176.X.XX.XXX:32400/library/metadata/39142/allLeaves?X-Plex-Token=XXXXXXXXXXXXXXXXXXXX
Sat, 02 Dec 2017 21:03:48 -0600api.php::fetchNumberedTVItem
Searching for season number 1 and episode number 1
Sat, 02 Dec 2017 21:03:48 -0600api.php::fetchInfo
Mods Found, fetching a numbered TV Item.
Sat, 02 Dec 2017 21:03:48 -0600api.php::fetchHubResults
Final results: []
Sat, 02 Dec 2017 21:03:48 -0600api.php::fetchMediaExtra
URL is http://176.X.XX.XXX:32400/library/metadata/39142?X-Plex-Token=XXXXXXXXXXXXXXXXXXXX
Sat, 02 Dec 2017 21:03:48 -0600api.php::fetchHubResults
Found an exact match: weeds
Sat, 02 Dec 2017 21:03:47 -0600api.php::fetchHubResults
URL is http://176.X.XX.XXX:32400/hubs/search?query=weeds&limit=30&X-Plex-Token=XXXXXXXXXXXXXXXXXXXX
Sat, 02 Dec 2017 21:03:47 -0600api.php::parsePlayCommand
NumberWordIn: ["first","pilot","second","third","last","final","latest","random"]
Sat, 02 Dec 2017 21:03:47 -0600api.php::parseApiCommand
Final media result: false
Sat, 02 Dec 2017 21:03:47 -0600api.php::parseApiCommand
Final params should be an action of play, a command of season 1 episode 1 weeds, a type of , and a control of .
Sat, 02 Dec 2017 21:03:47 -0600api.php::parseApiCommand
Action is currently play
Sat, 02 Dec 2017 21:03:47 -0600api.php::parseApiCommand
Raw speech is to play season 1 episode 1 weeds
Sat, 02 Dec 2017 21:03:47 -0600api.php::parseApiCommand
Full API.AI request: 
{
  "originalRequest": {
    "source": "google",
    "version": "2",
    "data": {
      "surface": {
        "capabilities": [
          {
            "name": "actions.capability.AUDIO_OUTPUT"
          },
          {
            "name": "actions.capability.MEDIA_RESPONSE_AUDIO"
          }
        ]
      },
      "inputs": [
        {
          "rawInputs": [
            {
              "query": "tell Flex TV to play season 1 episode 1 weeds",
              "inputType": "VOICE"
            }
          ],
          "arguments": [
            {
              "rawText": "to play season 1 episode 1 weeds",
              "textValue": "to play season 1 episode 1 weeds",
              "name": "trigger_query"
            },
            {
              "name": "3p_open_priority",
              "boolValue": true
            },
            {
              "rawText": "play",
              "textValue": "play",
              "name": "action"
            },
            {
              "rawText": "season 1 episode 1 weeds",
              "textValue": "season 1 episode 1 weeds",
              "name": "command"
            }
          ],
          "intent": "actions.intent.MAIN"
        }
      ],
      "user": {
        "lastSeen": "2017-12-03T02:27:34Z",
        "accessToken": "a6b4d7b5acbaeece0d6eda0903afd9eb486ab26dc0",
        "locale": "en-US",
        "userId": "ABwppHF1KTrE8rfK3oqAW00EW1Y_0edQF9oiXa1FE1tSs_b8ZO17PbYbmZr1CV7TzEHGZPmiy5EJXuA"
      },
      "conversation": {
        "conversationId": "1512270226191",
        "type": "NEW"
      },
      "availableSurfaces": [
        {
          "capabilities": [
            {
              "name": "actions.capability.AUDIO_OUTPUT"
            },
            {
              "name": "actions.capability.SCREEN_OUTPUT"
            }
          ]
        }
      ]
    }
  },
  "id": "9ab2c0e3-35ec-4373-b8a1-e883199b616b",
  "timestamp": "2017-12-03T03:03:46.402Z",
  "lang": "en-us",
  "result": {
    "source": "agent",
    "resolvedQuery": "to play season 1 episode 1 weeds",
    "action": "Playmedia",
    "parameters": {
      "action": "play",
      "command": "season 1 episode 1 weeds"
    },
    "contexts": [
      {
        "name": "promptfortitle",
        "parameters": {
          "command": "season 1 episode 1 weeds",
          "command.original": "season 1 episode 1 weeds",
          "action.original": "play",
          "action": "play"
        },
        "lifespan": 3
      },
      {
        "name": "waitforplayer",
        "parameters": {
          "command": "season 1 episode 1 weeds",
          "command.original": "season 1 episode 1 weeds",
          "action.original": "play",
          "action": "play"
        },
        "lifespan": 2
      },
      {
        "name": "google_assistant_welcome",
        "parameters": {
          "command": "season 1 episode 1 weeds",
          "command.original": "season 1 episode 1 weeds",
          "action.original": "play",
          "action": "play"
        },
        "lifespan": 5
      },
      {
        "name": "yes",
        "parameters": {
          "command": "season 1 episode 1 weeds",
          "command.original": "season 1 episode 1 weeds",
          "action.original": "play",
          "action": "play"
        },
        "lifespan": 3
      },
      {
        "name": "google_assistant_input_type_voice",
        "parameters": {
          "command": "season 1 episode 1 weeds",
          "command.original": "season 1 episode 1 weeds",
          "action.original": "play",
          "action": "play"
        }
      },
      {
        "name": "actions_capability_audio_output",
        "parameters": {
          "command": "season 1 episode 1 weeds",
          "command.original": "season 1 episode 1 weeds",
          "action.original": "play",
          "action": "play"
        }
      },
      {
        "name": "playmedia-followup",
        "parameters": {
          "command": "season 1 episode 1 weeds",
          "command.original": "season 1 episode 1 weeds",
          "action.original": "play",
          "action": "play"
        },
        "lifespan": 2
      },
      {
        "name": "actions_capability_media_response_audio",
        "parameters": {
          "command": "season 1 episode 1 weeds",
          "command.original": "season 1 episode 1 weeds",
          "action.original": "play",
          "action": "play"
        }
      }
    ],
    "metadata": {
      "matchedParameters": [
        {
          "dataType": "@Triggers",
          "name": "action",
          "value": "$action"
        },
        {
          "dataType": "@sys.any",
          "name": "command",
          "value": "$command"
        }
      ],
      "intentName": "PlayMedia",
      "intentId": "b8151275-afad-4f3f-bd9a-3a17f390cb75",
      "webhookUsed": "true",
      "webhookForSlotFillingUsed": "false",
      "nluResponseTime": 79
    },
    "fulfillment": {
      "messages": [
        []
      ]
    },
    "score": 1
  },
  "status": {
    "code": 200,
    "errorType": "success"
  },
  "sessionId": "1512270226191"
}
```

Let me know if I can provide anything else to help. Thanks!

**Comment by trippinshrumes** [comment]
So I applied the fix here for the test button

https://github.com/d8ahazard/Phlex/pull/228

Now I'm able to connect to sonarr, couchpotato, both plex servers.

I'm using a roku, it is detected by phlex. No matter the command I give it nothing works. 

The problematic error is this little piece right here

Unexpected HTTP code: 500, URL: https://176-9-36-199.e545575518324598890dd4b350dd314f.plex.direct:32400/player/playback/pause?type=video&commandID=0&X-Plex-Product=Phlex&X-Plex-Version=3.9.1&X-Plex-Client-Identifier=78fce331f37eeae20eaa39f4&X-Plex-Platform=Web&X-Plex-Platform-Version=1.0.0&X-Plex-Device=PhlexWeb&X-Plex-Device-Name=Phlex&X-Plex-Device-Screen-Resolution=1520x707,1680x1050,1920x1080&X-Plex-Token=48wBAfrwhxKR7T1ooTzY&X-Plex-Target-Client-Identifier=0763e79bc0c9c2f555abb30209e274a1&X-Plex-Token=48wBAfrwhxKR7T1ooTzY

On a side note, I did try disabling plex.direct again, no servers appear for playback servers, although test says it connected to the last. when I ask flex to do things it responds as if it's done it and has the same error.
